common/src/main/java/com/microsoft/identity/common/internal/msafederation/FederatedSignInProviderName.kt

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.msafederation
+
+/**
+ * Enum class for Federated Sign In Provider Name like Google, Apple
+ * Currently only Google is supported.
+ */
+enum class FederatedSignInProviderName(private val idProviderName: String) {
+    GOOGLE("google.com"),
+    APPLE("apple.com"); // would be used later
```

Comment:
We plan to use apple on Android?

Reply:
It has been part of discussions, but it may or may not be done based on business needs.

But there won't be sign in with apple using native app buttons so, no need to have have an API exposed for oneauth today or later.